testdata/p4_16_samples/pna-dpdk-add_on_miss0.py

```diff
@@ -16,170 +16,168 @@
-        
+        pkt_in = ptfutils.simple_tcp_packet(
+            eth_src=in_smac,
+            eth_dst=in_dmac,
+            ip_src=ip_src_addr,
+            ip_dst=ip_dst_addr,
+            tcp_sport=sport,
+            tcp_dport=dport,
+        )
+
         # Send in a first packet that should experience a miss on
         # table ct_tcp_table, cause a new entry to be added by the
         # data plane with a 30-second expiration time, and be
         # forwarded with a change to its source MAC address that the
         # add_on_miss0.p4 program uses to indicate that a miss
         # occurred.
-        logging.info("Sending packet #1")
-        tu.send_packet(self, ig_port, pkt_in)
+        bt.testutils.log.info("Sending packet #1")
+        ptfutils.send_packet(self, ig_port, pkt_in)
         first_pkt_time = time.time()
 
     def verifyPackets(self):
-        in_dmac = 'ee:30:ca:9d:1e:00'
-        in_smac = 'ee:cd:00:7e:70:00'
-        ip_src_addr = '1.1.1.1'
-        ip_dst_addr = '2.2.2.2'
+        in_dmac = "ee:30:ca:9d:1e:00"
+        in_smac = "ee:cd:00:7e:70:00"
+        ip_src_addr = "1.1.1.1"
+        ip_dst_addr = "2.2.2.2"
         sport = 59597
         dport = 7503
         # add_on_miss0.p4 replaces least significant 8 bits of source
         # MAC address with 0xf1 on a hit of table ct_tcp_table, or
         # 0xa5 on a miss.
-        out_smac_for_miss = in_smac[:-2] + 'a5'
+        out_smac_for_miss = in_smac[:-2] + "a5"
         eg_port = 1
-        exp_pkt_for_miss = \
-            tu.simple_tcp_packet(eth_src=out_smac_for_miss, eth_dst=in_dmac,
-                                 ip_src=ip_src_addr, ip_dst=ip_dst_addr,
-                                 tcp_sport=sport, tcp_dport=dport)
+        exp_pkt_for_miss = ptfutils.simple_tcp_packet(
+            eth_src=out_smac_for_miss,
+            eth_dst=in_dmac,
+            ip_src=ip_src_addr,
+            ip_dst=ip_dst_addr,
+            tcp_sport=sport,
+            tcp_dport=dport,
+        )
 
         # Check hit not tested for now for simplicity
-        # out_smac_for_hit = in_smac[:-2] + 'f1'
+        # out_smac_for_hit = in_smac[:-2] + "f1"
         # exp_pkt_for_hit = \
-        #     tu.simple_tcp_packet(eth_src=out_smac_for_hit, eth_dst=in_dmac,
+        #     ptfutils.simple_tcp_packet(eth_src=out_smac_for_hit, eth_dst=in_dmac,
         #                          ip_src=ip_src_addr, ip_dst=ip_dst_addr,
         #                          tcp_sport=sport, tcp_dport=dport)
 
-        
-        tu.verify_packets(self, exp_pkt_for_miss, [eg_port])
-        logging.info("    packet experienced a miss in ct_tcp_table as expected")
+        ptfutils.verify_packets(self, exp_pkt_for_miss, [eg_port])
+        bt.testutils.log.info("packet experienced a miss in ct_tcp_table as expected")
+
+        bt.testutils.log.info("Attempting to delete all entries in ipv4_host")
+        # TODO: This code does not seem functional?
```

Comment:
@jafingerhut I am unable to delete table entries on the P4Runtime interface for the DPDK SoftNIC. Even when I read the entries from the table, then try to delete those exact same entries I am getting a "not found" error. Is this a known issue?

The P4Runtme shell code also has the same issue, but it was never uncovered because it was only ever used on tables with no entries.

Reply:
I can double-check, but if I recall correctly, the current DPDK implementation does not support adding or deleting entries from the control plane specifically for add-on-miss tables.  For add-on-miss tables, it only supports adding and deleting entries from the data plane.

Reply:
Ah good to know. Strangely enough I am able to add entries, but I can not remove them.